pkg/kubelet/kubelet_node_status.go

```diff
@@ -394,6 +404,10 @@ func (kl *Kubelet) tryUpdateNodeStatus(tryNumber int) error {
 	if err != nil {
 		return err
 	}
+	if len(updatedNode.Status.Conditions) > 0 {
+		lastHeartbeatTime = updatedNode.Status.Conditions[0].LastHeartbeatTime
```

Comment:
same here, don't grab the timestamp of the first condition. an alternative could be to grab the latest timestamp of all the conditions that provide a non-zero heartbeat timestamp

Reply:
ok